### PR TITLE
Add updated secret model for secret usage restrictions

### PR DESCRIPTION
**Pull request by hamir-suspect**
This makes it possible for orgs that have this feature to manage secret restrictions with with api.

### CODE REVIEW

File: semaphore/secrets.v1beta.proto [hunk 1]
```diff
@@ -13,12 +13,23 @@ service SecretsApi {
...
 
     int64 create_time = 3; // unix timestamp of secret creation
     int64 update_time = 4; // unix timestamp of last update
+
+    SecretLevel level = 5;
+
+    string created_by = 6;
+    string updated_by = 7;
+    CheckoutMetadata last_checkout = 8;
+    int64 checkout_at = 9;
```
[the remark this answers is not in the document]
Comment (hamir-suspect) [reply semaphore/secrets.v1beta.proto:32]: `checkout_at ` <> `last_checkout_at`
checkout_at -> [checkout metadata](https://github.com/semaphoreci/api/pull/33/files#diff-56eafcb24d6d3b3b7a7f0d8e8d863d3f392e2f83a4d1ff263c92538625e28b3bR80)

[the remark this answers is not in the document]
Comment (radwo) [reply semaphore/secrets.v1beta.proto:32]: I believe that Igor is asking if this field is a timestamp of the last checkout, the first one, or maybe any checkout.

[the remark this answers is not in the document]
Comment (hamir-suspect) [reply semaphore/secrets.v1beta.proto:32]: Yes `checkout_at` is the last checkout time

[the remark this answers is not in the document]
Comment (radwo) [reply semaphore/secrets.v1beta.proto:32]: so let's call it `last_checkout_at` to be in line with `CheckoutMetadata last_checkout = 8;`

**Pull request closed without merging**